docs/theme.md

```diff
@@ -0,0 +1,138 @@
+# Theme API
+
+A theme in contexture-react is simply an object -- which we'll call a **theme object** for clarity -- that maps **theme keys** to React components. The theme object lives inside contexture-react's `ThemeContext`, which uses React context under the hood.
+
+Since they are really just identifiers for components, theme keys should follow the same naming conventions as React components themselves. For the purpose of consuming the API, we'll refer to a theme key together with the component it identifies as a **theme component**.
+
+## Using the Theme API
+
+To use **theme components** inside your own component definitions, your component must pull them from the theme object within contexture-react's `ThemeContext`.
+
+There are two ways to get the theme from context: the `ThemeConsumer` component, and the `withTheme` function.
+
+### ThemeConsumer
+
+The `ThemeConsumer` helper component takes a render function as its child, and passes the **theme object** from context into that function. This example demonstrates how to create a basic IconButton component using the `Icon` and `Button` **theme components**, plus `ThemeConsumer`:
+
+```jsx
+// IconButton.js
+
+import React from 'react'
+import { ThemeConsumer } from 'contexture-react/src/utils/theme'
+
+let IconButton = ({ icon, children, ...props }) => (
+  <ThemeConsumer>
+    {theme =>
+      <theme.Button {...props}>
+        <theme.Icon icon={icon} />
+        {children}
+      </theme.Button>
+    }
+  </ThemeConsumer>
+)
+export default IconButton
+```
+
+### withTheme
+
+The `withTheme` higher-order component injects the theme object from context into its component argument as the `theme` prop. Here is the same IconButton example using `withTheme`:
+
+```jsx
+// IconButton.js
+
+import React from 'react'
+import { withTheme } from 'contexture-react/src/utils/theme'
+
+let IconButton = ({ theme, icon, children, ...props }) => (
+  <theme.Button {...props}>
+    <theme.Icon icon={icon} />
+    {children}
+  </theme.Button>
+)
+export default withTheme(IconButton)
+```
+
+`ThemeConsumer` and `withTheme` both accomplish the same thing in different ways. You're free to use whichever one you like best. 🙂
+
+
+## Default theme components
+
+The default theme in contexture-react is GreyVest. The full list of **theme components** included in GreyVest is shown in the table below.
+
+| Key | Expected props | Notes |
```

Comment:
I don't think organizing this table alphabetically is all that helpful - let's try to group them into logical sections (so that, for example, ExpandableTable and Table are next to each other). I could see us having a section for inputs (checkbox, input, radio list, tags input, etc), containers (modal, popover, box), and maybe another section. Things that are hard to clearly put into one section or another are definitely things that should be scrutinized